server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2247,7 +2247,7 @@ private void updateTemplate(long templateId, String fileExtension, long zoneId,
     }
 
     private void updateDirectDownloadTemplate(long templateId, long zoneId) {
-        TemplateDataStoreVO templateStore = _tmplStoreDao.findByTemplateZone(templateId, zoneId, DataStoreRole.Image);
+        TemplateDataStoreVO templateStore = _tmplStoreDao.findByTemplate(templateId, DataStoreRole.Image);
```

Comment:
So we are updating every reference in every zone for the template?

Reply:
@nvazquez as we do not store `store_id` in `cloud.template_store_ref` table for direct download template, we cannot find the entry in the table for a zone while update. Continued with same behaviour from user direct download templates. Do we need to change this for system direct download templates?